coorx/coordinates.py

```diff
@@ -25,7 +25,8 @@ class PointArray:
     def __init__(self, coordinates, system: CoordSysOrStr = None, graph: StrOrNone = None):
         coord_arr, source_system = self._interpret_input(coordinates)
 
-        assert coord_arr.dtype is not np.dtype(object)
+        if coord_arr.dtype is np.dtype(object):
```

Comment:
`np.issubdtype(coord_arr.dtype, np.number)`
This will catch others (like string types) as well.